app/src/main/java/uk/gov/onelogin/login/BioOptInViewModel.kt

```diff
@@ -5,12 +5,15 @@ import dagger.hilt.android.lifecycle.HiltViewModel
 import javax.inject.Inject
 import uk.gov.onelogin.login.biooptin.BiometricPreference
 import uk.gov.onelogin.login.biooptin.BiometricPreferenceHandler
+import uk.gov.onelogin.tokens.usecases.AutoInitialiseSecureStore
 
 @HiltViewModel
 class BioOptInViewModel @Inject constructor(
-    private val biometricPreferenceHandler: BiometricPreferenceHandler
+    private val biometricPreferenceHandler: BiometricPreferenceHandler,
+    private val autoInitialiseSecureStore: AutoInitialiseSecureStore
 ) : ViewModel() {
     fun setBioPreference(pref: BiometricPreference) {
         biometricPreferenceHandler.setBioPref(pref)
+        autoInitialiseSecureStore()
```

Comment:
why do we need this here?

Reply:
We initialise the SecureStore as soon as we have all the info needed. 

In the case when we have a fresh app install, we do not know the BiometricPreference. After they login, and they choose the BiometricPreference, then we have all the info we need to initialise the secure store. So we do it here.

If they restart the app, and we have the Biometric Preference at the app startup, then we initialise there.